### PR TITLE
Updated for the SDL3 RW -> IO changes

### PR DESCRIPTION
**Pull request by slouken**

### CODE REVIEW

**Comment by sezero** [comment]
Unwanted permission changes:
```
 26 files changed, 1076 insertions(+), 1078 deletions(-)
 mode change 100644 => 100755 include/SDL3_image/SDL_image.h
 mode change 100644 => 100755 src/IMG.c
 mode change 100644 => 100755 src/IMG.h
 mode change 100644 => 100755 src/IMG_ImageIO.m
 mode change 100644 => 100755 src/IMG_WIC.c
 mode change 100644 => 100755 src/IMG_avif.c
 mode change 100644 => 100755 src/IMG_bmp.c
 mode change 100644 => 100755 src/IMG_gif.c
 mode change 100644 => 100755 src/IMG_jpg.c
 mode change 100644 => 100755 src/IMG_jxl.c
 mode change 100644 => 100755 src/IMG_lbm.c
 mode change 100644 => 100755 src/IMG_pcx.c
 mode change 100644 => 100755 src/IMG_png.c
 mode change 100644 => 100755 src/IMG_pnm.c
 mode change 100644 => 100755 src/IMG_qoi.c
 mode change 100644 => 100755 src/IMG_stb.c
 mode change 100644 => 100755 src/IMG_svg.c
 mode change 100644 => 100755 src/IMG_tga.c
 mode change 100644 => 100755 src/IMG_tif.c
 mode change 100644 => 100755 src/IMG_webp.c
 mode change 100644 => 100755 src/IMG_xcf.c
 mode change 100644 => 100755 src/IMG_xpm.c
 mode change 100644 => 100755 src/IMG_xv.c
 mode change 100644 => 100755 src/IMG_xxx.c
 mode change 100644 => 100755 test/main.c
```

File: include/SDL3_image/SDL_image.h [hunk 1]
```diff
@@ -210,7 +210,7 @@ extern DECLSPEC void SDLCALL IMG_Quit(void);
  * by calling: SDL_SetSurfaceColorKey(image, SDL_RLEACCEL,
  * image->format->colorkey);
  *
- * If `freesrc` is SDL_TRUE, the RWops will be closed before returning,
+ * If `closeio` is SDL_TRUE, the RWops will be closed before returning,
  * whether this function succeeds or not. SDL_image reads everything it needs
  * from the RWops during this call in any case.
```
Comment (sezero) [inline include/SDL3_image/SDL_image.h:215]: There are multitudes of occurrences of `RWops` in there which I guess we should change to `IOStream`

**Comment by sezero** [comment]
src/SDL_image.sym and test/main.c still have references to _RW procuderes, e.g. IMG_SaveJPG_RW which I think is the reason for CI failures

**Comment by sezero** [comment]
src/IMG_ImageIO.m isn't updated, hence the Mac CI failure

**Comment by slouken** [comment]
Thanks for the review. I fixed the permissions and it looks like CI is succeeding. Can you take one last look as a sanity check?

**Comment by sezero** [comment]
Missed one permission change: ` mode change 100644 => 100755 src/IMG.h`

Missed changes:
```diff
diff --git a/CHANGES.txt b/CHANGES.txt
index c485d4c..5084189 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -3,2 +3,2 @@
  * Added AVIF save support:
-    IMG_SaveAVIF() and IMG_SaveAVIF_RW()
+    IMG_SaveAVIF() and IMG_SaveAVIF_IO()
diff --git a/README.txt b/README.txt
index 4a5ebc6..9fe9780 100644
--- a/README.txt
+++ b/README.txt
@@ -14,8 +14,8 @@ API:
 or
-	SDL_Surface *IMG_Load_RW(SDL_RWops *src, SDL_bool freesrc);
+	SDL_Surface *IMG_Load_IO(SDL_IOStream *src, SDL_bool closeio);
 or
-	SDL_Surface *IMG_LoadTyped_RW(SDL_RWops *src, SDL_bool freesrc, char *type);
+	SDL_Surface *IMG_LoadTyped_IO(SDL_IOStream *src, SDL_bool closeio, char *type);
 
 where type is a string specifying the format (i.e. "PNG" or "pcx").
-Note that IMG_Load_RW cannot load TGA images.
+Note that IMG_Load_IO cannot load TGA images.
 
diff --git a/src/IMG_xpm.c b/src/IMG_xpm.c
index 0f14d73..6c7715b 100644
--- a/src/IMG_xpm.c
+++ b/src/IMG_xpm.c
@@ -996,3 +996,3 @@ do {                            \
 
-/* read XPM from either array or RWops */
+/* read XPM from either array or IOStream */
 static SDL_Surface *load_xpm(char **xpm, SDL_IOStream *src, SDL_bool force_32bit)
@@ -1170,3 +1170,3 @@ done:
 
-/* Load a XPM type image from an RWops datasource */
+/* Load a XPM type image from an IOStream datasource */
 SDL_Surface *IMG_LoadXPM_IO(SDL_IOStream *src)
diff --git a/src/IMG_xxx.c b/src/IMG_xxx.c
index 6c23bc2..b3c1d4c 100644
--- a/src/IMG_xxx.c
+++ b/src/IMG_xxx.c
@@ -44,3 +44,3 @@ int IMG_isXXX(SDL_IOStream *src)
 
-    SDL_SeekIO(src, start, SDL_RW_SEEK_SET);
+    SDL_SeekIO(src, start, SDL_IO_SEEK_SET);
     return is_XXX;
@@ -50,3 +50,3 @@ int IMG_isXXX(SDL_IOStream *src)
 /* Remember to declare this procedure in IMG.h . */
-SDL_Surface *IMG_LoadXXX_RW(SDL_IOStream *src)
+SDL_Surface *IMG_LoadXXX_IO(SDL_IOStream *src)
 {
@@ -57,3 +57,3 @@ SDL_Surface *IMG_LoadXXX_RW(SDL_IOStream *src)
     if (!src) {
-        /* The error message has been set in SDL_RWFromFile */
+        /* The error message has been set in SDL_IOFromFile */
         return NULL;
@@ -66,3 +66,3 @@ SDL_Surface *IMG_LoadXXX_RW(SDL_IOStream *src)
     if (error) {
-        SDL_SeekIO(src, start, SDL_RW_SEEK_SET);
+        SDL_SeekIO(src, start, SDL_IO_SEEK_SET);
         if (surface) {
@@ -89,3 +89,3 @@ int IMG_isXXX(SDL_IOStream *src)
 
-SDL_Surface *IMG_LoadXXX_RW(SDL_IOStream *src)
+SDL_Surface *IMG_LoadXXX_IO(SDL_IOStream *src)
 {
```

**Comment by slouken** [comment]
Thank you!

**Review by sezero — APPROVED** [review]

**Pull request merged**